pkg/pool/pool.go

```diff
@@ -715,6 +736,73 @@ func (p *pool) getPDDeviation() int {
 	return deviationPrefix * NumIPv4AddrPerPrefix
 }
 
+// calculateSecondaryIPDeviation calculates the deviation required to meet the desired state for secondary IP mode
+// Returns a number of IPv4 addresses by taking into account the MinIPTarget and WarmIPTarget
+func (p *pool) calculateSecondaryIPDeviation() int {
+	numWarmResources := numResourcesFromMap(p.warmResources)
+	numUsedResources := len(p.usedResources)
+	numAssignedResources := numUsedResources + numWarmResources + p.pendingCreate + len(p.coolDownQueue)
+
+	// warm pool is in draining state, set targets to zero
+	if p.warmPoolConfig.DesiredSize == 0 {
+		p.log.V(1).Info("DesiredSize is zero, warmPool is in draining state")
+		p.warmPoolConfig.WarmIPTarget = 0
+		p.warmPoolConfig.MinIPTarget = 0
+		p.warmPoolConfig.WarmPrefixTarget = 0
+	}
+
+	isMinIPTargetInvalid := p.warmPoolConfig.MinIPTarget < 0
```

Comment:
Why should we have invalid values be passed with p.warmPoolConfig? We should make sure our managed fields valid before set them up.

Reply:
Indeed the invalid negative values should never make their way into this code logic sequence. It seems redundant to recheck this, but I think its a reasonable defensive coding precaution to avoid unintended behaviour. 

Another suggestion I considered was to change the type of IP target configurations e.g. `MinIPTarget` from `int` to `uint` which only allows positive values but this would require refactoring in multiple places and I wanted to keep this change minimal.

Reply:
@haouc I am resolving the conversation to unblock the PR. Let me know if any concerns.